sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -462,6 +462,9 @@ export class EventHubReceiver extends LinkEntity {
         );
       }
       if (shouldReopen) {
+        // Remove the existing receiver from the internal cache.
+        // A new one will be created with a different name (newName: true).
+        this._deleteFromCache();
```

Comment:
Consider a case where detached() fails (even after the retries), then we have the receiver deleted from the cache for ever.

Is that expected?

What **is** the expectation when detached fails anyway?

Reply:
I think there could be an argument for deleting the receiver if it can't be re-initialized.

Looking at the `disconnected` function in `connectionContext`:
https://github.com/Azure/azure-sdk-for-js/blob/%40azure/event-hubs_2.1.1/sdk/eventhub/event-hubs/src/connectionContext.ts#L189-L198

All receivers in the cache will have `detached` called on them assuming they aren't already connecting. If we had called `detached` somewhere else in our code, and the `_init` retries ultimately failed such that the receiver is no longer trying to connect, it's possible that the old receiver would attempt to connect again since it would still be in the cache.

Is this what we want to happen? I'm not sure yet. Related to this discussion, I think we should call the user's error handler if the `_init` retries do fail so that a user can decide if they want to close the receiver down and start a new one.

Interestingly, the above scenario could present itself as a receive operation that stops receiving events (stuck attempting to recreate a connection) due to connection issues that suddenly starts working again when a different receiver or the connection receives a `disconnect` error (e.g. ECONNRESET)